test/aries-js-worker/test/issuecredential/issuecredential.js

```diff
@@ -0,0 +1,132 @@
+        issuer = didexClient.agent2
+    })
+
+    after(() => {
+        didexClient.destroy()
+    })
+
+    it("Holder requests credential from the Issuer", async function() {
+        let conn = await connection(holder, connections[0])
+        return holder.issuecredential.sendRequest({
+            my_did: conn.MyDID,
+            their_did: conn.TheirDID,
+            request_credential: {}
+        })
+    })
+
+    it("Issuer accepts request and sends credential to the Holder", async function() {
+        let action = await getAction(issuer)
+        return issuer.issuecredential.acceptRequest({
+            piid: action.piid,
+            issue_credential: {
+                "credentials~attach": [{
+                    "lastmod_time": "0001-01-01T00:00:00Z",
+                    "data": {
+                        "json": {
+                            "@context": [
+                                "https://www.w3.org/2018/credentials/v1",
+                                "https://www.w3.org/2018/credentials/examples/v1"
+                            ],
+                            "credentialSubject": {
+                                "ID": "SubjectID"
+                            },
+                            "id": "http://example.edu/credentials/1872",
+                            "issuanceDate": "2010-01-01T19:23:24Z",
+                            "issuer": {
+                                "id": "did:example:76e12ec712ebc6f1c221ebfeb1f",
+                                "name": "Example University"
+                            },
+                            "referenceNumber": 83294847,
+                            "type": [
+                                "VerifiableCredential",
+                                "UniversityDegreeCredential"
+                            ]
+                        }
+                    }
+                }]
+            }
+        })
+    })
+
+    const credentialName = "license"
+
+    it("Holder accepts credential", async function() {
+        let action = await getAction(holder)
+        return holder.issuecredential.acceptCredential({
+            piid: action.piid,
+            names: [credentialName],
+        })
+    })
+
+    it("Checks credential", async function() {
+        return getCredential(holder, credentialName)
+    })
+})
+
+const retries = 10;
+
+async function getAction(agent) {
+    for (let i = 0; i < retries; i++) {
+        let resp = await agent.issuecredential.actions()
+        if (resp.actions.length > 0) {
+            return resp.actions[0]
+        }
+        await sleep(1000);
+    }
+
+    throw new Error("no action")
+}
+
+async function getCredential(agent, name) {
+    for (let i = 0; i < retries; i++) {
+        try {
+            return await agent.verifiable.getCredentialByName({
+                name: name
+            })
+        } catch (e) {
+            console.warn(e.message)
+        }
+
+        await sleep(1000);
```

Comment:
sleeps make me worried about test fragility

Reply:
I don't know when the message will be saved. There is no way to listen to the post `StateMsg` event.  Or there is?

Reply:
I guess we should consider adding that capability :).

Reply:
I will check what we can do. Only the loop is the solution. Also, I saw we did it before.

Reply:
sure - loop will help.